MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/TextureRender.java

```diff
@@ -296,6 +270,38 @@ private int compileShader(int type, String shaderSrc) {
         return shader;
     }
 
+    public void setUMatrix(float scaleFactor, float xTranslate, float yTranslate) {
+        // Scale and translate matrix for auto-crop
+        Matrix.translateM(uMatrix, 0, xTranslate, yTranslate, 0.0f);
+        Matrix.scaleM(uMatrix, 0, scaleFactor, scaleFactor, 1.0f);
+
+        // Set up rotation matrix if the camera orientation is greater than 0 degrees
+        if (videoRotation > 0) {
+            Log.d(TAG, "rotation: " + videoRotation);
+            Matrix.rotateM(
+                    uMatrix,
+                    /* rmOffset = */ 0,
+                    /* a = */ -videoRotation,  // Original video rotation is stored clockwise
+                    /* x = */ 0,
+                    /* y = */ 0,
+                    /* z = */ 1
+            );
+        }
+
+        // TODO: Switch to VBOs and VAOs
+        uMatrixHandle = glGetUniformLocation(program, "uMatrix");
+        glUniformMatrix4fv(
+                uMatrixHandle,
+                /* count = */ 1,
+                /* transpose = */ false,
+                uMatrix,
+                /* offset = */ 0
+        );
+        if (glGetError() != 0) {
+            throw new RuntimeException("Failed to get matrix");
```

Comment:
Maybe `Log.e` instead? In the worst case it will result in visual artifacts, maybe we should not crash over it?

Reply:
It seems that you still have a lot of these RuntimeExceptions thrown when GL returns errors, is that deliberate?

Reply:
No, these should have been changed. I didn't see these comments the first time around--sorry about that!

Reply:
No problem!